### PR TITLE
Log stdout into file

### PR DESCRIPTION
**Pull request by charmoniumQ**

### CODE REVIEW

**Review by apache8080 — COMMENTED** [review]
If the changes to the config files were for testing purposes, please revert them back to whats on master so its cleaner.

**Comment by charmoniumQ** [comment]
> If the changes to the config files were for testing purposes, please revert them back to whats on master so its cleaner.

Resolved.

File: runner/runner/main.py [hunk 1]
```diff
@@ -116,21 +117,36 @@ def load_native(config: Mapping[str, Any]) -> None:
...
     )
-    subprocess_run(
-        command_lst_sbst,
-        env_override=dict(ILLIXR_DATA=str(data_path), ILLIXR_DEMO_DATA=str(demo_data_path), KIMERA_ROOT=config["loader"]["kimera_path"]),
-        check=True,
+    log_stdout_str = config["loader"].get("log_stdout", None)
+    log_stdout_ctx: ContextManager[Optional[BinaryIO]] = cast(ContextManager[Optional[BinaryIO]],
+        open(log_stdout_str, "wb")
+        if log_stdout_str is not None
+        else noop_context(None)
```
Comment (e3m3) [inline runner/runner/main.py:141]: The `if-else` if for the value of the `open(...)` right? Using some parenthesis and white space might make this easier to parse.

**Review by e3m3 — DISMISSED** [review]
Lgtm other than the minor formatting comment

File: runner/runner/main.py (hunk 1 above)
Comment (charmoniumQ) [reply runner/runner/main.py]: I slapped some parens on there and used Black, which emitted:
```
    log_stdout_ctx = cast(
        ContextManager[Optional[BinaryIO]],
        (
            open(log_stdout_str, "wb")
            if (log_stdout_str is not None)
            else noop_context(None)
        ),
    )
```

Is this more readable?

**Review by mhuzai — DISMISSED** [review]

File: runner/runner/main.py (hunk 1 above)
Comment (charmoniumQ) [reply runner/runner/main.py]: Bump

File: runner/runner/main.py [hunk 2]
```diff
@@ -116,21 +115,48 @@ def load_native(config: Mapping[str, Any]) -> None:
...
+        KIMERA_ROOT=config["loader"]["kimera_path"],
+    )
+    env_list = [
+        f"{shlex.quote(var)}={shlex.quote(val)}" for var, val in env_override.items()
+    ]
+    actual_cmd_list = list(
         flatten1(
             replace_all(
-                unflatten(shlex.split(command_str)),
-                {("%a",): main_cmd_lst, ("%b",): [shlex.quote(shlex.join(main_cmd_lst))]},
```
Comment (charmoniumQ) [inline runner/runner/main.py:125]: This change conflicts with the documentation in #215. It does not induce an ordering; changes are required in whichever is not merged first.

`%a` -> `$cmd` or `$env_cmd`
`%b` -> `$quoted_cmd`

**Review by e3m3 — APPROVED** [review]

File: runner/runner/main.py (hunk 2 above)
Comment (e3m3) [reply runner/runner/main.py:125]: This PR can be merged first. Which page/section is the conflict in?

Comment (charmoniumQ) [reply runner/runner/main.py:125]: `docs/building_illixr.md`

- `$cmd` is the pure cmd, argument for argument.
- `$quoted_cmd` is the cmd as one argument.
- `$env_cmd` is like `$cmd`, but it sets env-vars explicitly `env A=B C=D ... $cmd`. The runner already sets the env-vars in the process of `command`, but this is useful if you want to run outside the runner (you'll need the env-vars explicitly set).

**Review by e3m3 — APPROVED** [review]

**Review by mhuzai — DISMISSED** [review]
I'm guessing the rebase will dismiss this. :-/

**Review by mhuzai — APPROVED** [review]

**Review by e3m3 — APPROVED** [review]

**Pull request merged**